adapters/sidekiq/lib/opentelemetry/adapters/sidekiq/middlewares/server/tracer_middleware.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Sidekiq
+      module Middlewares
+        module Server
+          class TracerMiddleware
+            def call(_worker, msg, _queue)
+              parent_context = OpenTelemetry.propagation.text.extract(msg)
+              tracer.in_span(
+                msg['wrapped']&.to_s || msg['class'],
+                attributes: {
+                  'job_id' => msg['jid'],
+                  'messaging.destination' => msg['queue'],
+                  'created_at' => msg['created_at'],
+                  'enqueued_at' => msg['enqueued_at'],
```

Comment:
I feel like these are better recorded as events rather than tags (especially since attribute values can’t be time stamps). https://github.com/open-telemetry/opentelemetry-ruby/blob/cd4b44daa4453e78eb4e2c2ced80ee84f680eb65/api/lib/opentelemetry/trace/span.rb#L88

E.g.:
```ruby
span.add_event(name: 'enqueued_at', timestamp: msg['enqueued_at'])
```